src/Integration/LocalServices/ProjectSystemFilter.cs

```diff
@@ -34,48 +34,35 @@ public ProjectSystemFilter(IHost host)
 
         #region IProjectSystemFilter
 
-        public bool IsAccepted(DteProject dteProject)
+        public bool IsAccepted(DteProject project)
         {
-            if (dteProject == null)
+            if (project == null)
             {
-                throw new ArgumentNullException(nameof(dteProject));
+                throw new ArgumentNullException(nameof(project));
             }
 
-            var projectName = dteProject.Name;
-            var hierarchy = this.projectSystem.GetIVsHierarchy(dteProject);
+            var projectName = project.Name;
+            var hierarchy = this.projectSystem.GetIVsHierarchy(project);
             var propertyStorage = hierarchy as IVsBuildPropertyStorage;
 
             if (hierarchy == null || propertyStorage == null)
             {
-                throw new ArgumentException(Strings.ProjectFilterDteProjectFailedToGetIVs, nameof(dteProject));
+                throw new ArgumentException(Strings.ProjectFilterDteProjectFailedToGetIVs, nameof(project));
             }
 
             // Accept only supported languages
-            var language = Language.ForProject(dteProject);
-            if (language == null || !language.IsSupported)
+            if (IsNotSupportedProject(project))
             {
                 return false;
             }
 
-            // General exclusions
-            // If exclusion property is set, this takes precedence
-            bool? sonarExclude = GetPropertyBool(propertyStorage, Constants.SonarQubeExcludeBuildPropertyKey);
-            if (sonarExclude.HasValue)
+            if (IsExcludedViaProjectProperty(propertyStorage))
             {
-                return !sonarExclude.Value;
-            }
-
-            // Ignore test projects
-            // If specifically marked with test project property, use that to specifiy if test project or not
-            bool? sonarTest = GetPropertyBool(propertyStorage, Constants.SonarQubeTestProjectBuildPropertyKey);
-            if (sonarTest.HasValue)
-            {
-                return !sonarTest.Value;
+                return false;
```

Comment:
I don't think that either the previous and proposed versions replicates what the scanner does :-). We can discuss whether what the scanner does is correct or not, but that's a separate issue: the current goal is to make the connected mode behave the same as the scanner.

The scanner's processing logic is in the [targets file](https://github.com/SonarSource-VisualStudio/sonar-msbuild-runner/blob/master/SonarQube.MSBuild.Tasks/Targets/SonarQube.Integration.targets). For the current purposes, the important bit is whether we should run Roslyn code analysis or not: lines 405 to 446.

The scanner only runs code analysis if the project is not excluded and not a test project
i.e. the scanner won't run code analysis if it thinks the project is a test project, even if _SonarQubeExclude_=false.

The previous implementation doesn't match the scanner because if _SonarQubeExclude_ was set it would use just that value.
The proposed implementation doesn't match the scanner because it continues checking when _SonarQubeTestProject_ is explicitly set to false.